modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -1612,75 +1612,172 @@ public static boolean isAnyType(Schema schema) {
-                target.setHasValidation(true);
+        Integer minLength = schema.getMinLength();
+        if(minLength!=null) vSB.withMinLength();
+
+        Integer maxLength = schema.getMaxLength();
+        if(maxLength!=null) vSB.withMaxLength();
+
+        String pattern = schema.getPattern();
+        if(pattern!=null) vSB.withPattern();
+
+        BigDecimal multipleOf = schema.getMultipleOf();
+        if(multipleOf!=null) vSB.withMultipleOf();
+
+        BigDecimal minimum = schema.getMinimum();
+        if(minimum!=null) vSB.withMinimum();
+
+        BigDecimal maximum = schema.getMaximum();
+        if(maximum!=null) vSB.withMaximum();
+
+        Boolean exclusiveMinimum = schema.getExclusiveMinimum();
+        if(exclusiveMinimum!=null) vSB.withExclusiveMinimum();
+
+        Boolean exclusiveMaximum = schema.getExclusiveMaximum();
+        if(exclusiveMaximum!=null) vSB.withExclusiveMaximum();
+
+        Set<String> setValidations = vSB.build();
+
+        if (isArraySchema(schema)) {
+            logWarnMessagesForIneffectiveValidations(setValidations, schema, SchemaValidations.ARRAY_VALIDATIONS);
+            setArrayValidations(schema, target);
+        } else if (isTypeObjectSchema(schema)) {
+            logWarnMessagesForIneffectiveValidations(setValidations, schema, SchemaValidations.OBJECT_VALIDATIONS);
+            setObjectValidations(schema, target);
+        } else if (isStringSchema(schema)) {
+            setStringValidations(schema, target);
+            if (isDecimalSchema(schema)) {
+                setNumericValidations(schema, target);
+                logWarnMessagesForIneffectiveValidations(setValidations, schema, SchemaValidations.ARRAY_VALIDATIONS);
             }
+            else
+                logWarnMessagesForIneffectiveValidations(setValidations, schema, SchemaValidations.ARRAY_VALIDATIONS);
+
+        } else if (isNumberSchema(schema) || isIntegerSchema(schema)) {
+            setNumericValidations(schema, target);
+            logWarnMessagesForIneffectiveValidations(setValidations, schema, SchemaValidations.NUMERIC_VALIDATIONS);
+        } else if (isAnyType(schema)) {
+            // anyType can have any validations set on it
+            setArrayValidations(schema, target);
+            setObjectValidations(schema, target);
+            setStringValidations(schema, target);
+            setNumericValidations(schema, target);
+            logWarnMessagesForIneffectiveValidations(setValidations, schema, SchemaValidations.ALL_VALIDATIONS);
         }
+
+        if(!setValidations.isEmpty())
+            target.setHasValidation(true);
     }
 
-    private static void setArrayValidations(Integer minItems, Integer maxItems, Boolean uniqueItems, IJsonSchemaValidationProperties target) {
```

Comment:
How about reverting these setXValidations methods and passing in the inputs like we did before
- Please also add an early return in these helpers if all the inputs are null.

Your code:
- is extracting values twice